lib/streamlit/elements/time_widgets.py

```diff
@@ -206,17 +206,8 @@ def date_input(
 
         if single_value:
             value = [value]
-
-        date_input_proto = DateInputProto()
```

Comment:
This has just been moved, not changed. Setting `value` to a list of dates makes the error checking simpler so I put it above the error checking and grouped setting the proto logic below error checking.